src/gui/processing/qgsprocessingoutputdestinationwidget.cpp

```diff
@@ -728,4 +765,25 @@ void QgsProcessingLayerOutputDestinationWidget::dropEvent( QDropEvent *event )
   leText->setHighlighted( false );
 }
 
+bool QgsProcessingLayerOutputDestinationWidget::isTemporaryLayer( const QString &value )
+{
+  if ( value == QgsProcessing::TEMPORARY_OUTPUT || value.isEmpty() || mParameter->type() != QgsProcessingParameterFeatureSink::typeName() || !mParameter->supportsNonFileBasedOutput() )
+    return false;
```

Comment:
why return false when value == temporary_output?

Or maybe this method should be renamed to "isVectorMemoryLayer" ?

Reply:
...or when value is empty?